assets/js/googlesitekit/data/create-settings-store.js

```diff
@@ -169,12 +169,13 @@ export const createSettingsStore = ( type, identifier, datapoint, {
 		 */
 		*saveSettings() {
 			const registry = yield Data.commonActions.getRegistry();
-			const values = registry.select( STORE_NAME ).getSettings();
+			registry.dispatch( STORE_NAME ).clearError( 'saveSettings', [] );
 
+			const values = registry.select( STORE_NAME ).getSettings();
 			const { response, error } = yield fetchSaveSettingsStore.actions.fetchSaveSettings( values );
 			if ( error ) {
 				// Store error manually since saveSettings signature differs from fetchSaveSettings.
-				registry.dispatch( STORE_NAME ).receiveError( error, 'saveSettings' );
+				registry.dispatch( STORE_NAME ).receiveError( error, 'saveSettings', [] );
```

Comment:
There's a problem here which is unrelated to this issue, but was missed previously when introducing the dynamic error functionality: `receiveError` here should only be called if it's available on the store.

We can assume that that is the case in all other areas where we defined specific partial stores, but this here is a reusable function, and there could be a store that calls `createSettingsStore` but does not call `createErrorStore` for the same store - then this would fail. So we should first check if that action exists on the store and only call it then. Or we could do it the same way as in https://github.com/google/site-kit-wp/blob/1.19.0/assets/js/googlesitekit/data/create-fetch-store.js#L43, that way we don't need to do a check.

Reply:
Updated.